db/sql/src/tickets.rs

```diff
@@ -0,0 +1,2896 @@
+use async_stream::stream;
+use hopr_db_api::resolver::HoprDbResolverOperations;
+use hopr_db_api::tickets::AggregationPrerequisites;
+use std::cmp;
+use std::ops::Add;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use futures::TryStreamExt;
+use sea_orm::{ActiveModelTrait, ColumnTrait, EntityTrait, IntoActiveModel, QueryFilter, QueryOrder, Set};
+use sea_query::{Condition, Expr, IntoCondition};
+use tracing::{debug, error, info, trace};
+
+use hopr_crypto_types::prelude::*;
+use hopr_db_api::{
+    errors::Result,
+    info::DomainSeparator,
+    tickets::{ChannelTicketStatistics, HoprDbTicketOperations, TicketSelector},
+};
+use hopr_db_entity::ticket_statistics;
+use hopr_db_entity::{outgoing_ticket_index, ticket};
+use hopr_internal_types::prelude::*;
+use hopr_primitive_types::prelude::*;
+
+use crate::channels::HoprDbChannelOperations;
+use crate::db::HoprDb;
+use crate::errors::DbSqlError;
+use crate::errors::DbSqlError::LogicalError;
+use crate::info::HoprDbInfoOperations;
+use crate::{HoprDbGeneralModelOperations, OpenTransaction, OptTx, TargetDb};
+
+#[cfg(all(feature = "prometheus", not(test)))]
+use hopr_metrics::metrics::MultiGauge;
+
+#[cfg(all(feature = "prometheus", not(test)))]
+lazy_static::lazy_static! {
+    pub static ref METRIC_HOPR_TICKETS_INCOMING_STATISTICS: MultiGauge = MultiGauge::new(
+        "hopr_tickets_incoming_statistics",
+        "Ticket statistics for channels with incoming tickets.",
+        &["channel", "statistic"]
+    ).unwrap();
+}
+
+pub(crate) struct WrappedTicketSelector(pub(crate) TicketSelector);
```

Comment:
This is solely needed because of the `hopr_db_api` and `hopr_db_sql` crate split. You cannot implement a trait defined outside your crate for a type defined outside your crate. And since the `TicketSelector` is a `hopr_db_api` type, that just so needs to implement a `hopr_db_sql` specific implementation, the type is wrapped to allow the impl over that type.